ui/store/actions.js

```diff
@@ -812,13 +812,8 @@ export function txError(err) {
 }
 
 ///: BEGIN:ONLY_INCLUDE_IN(flask)
-export function removeSnapError(msgData) {
-  return (dispatch) => {
-    return promisifiedBackground
-      .removeSnapError(msgData)
-      .then(() => updateMetamaskStateFromBackground())
-      .then((newState) => dispatch(updateMetamaskState(newState)));
-  };
```

Comment:
ah i thought the front-end needed to call `updateMetamaskState` after every background call to update the whole ui's state.